internals/src/error.rs

```diff
@@ -6,6 +6,11 @@
 //! Error handling macros and helpers.
```

Comment:
Not that it matters since this is draft but did you mean to use the newer form `error.rs` and `error/` instead of `error/mod.rs`?

Reply:
Honestly, I don't care about these forms. I normally use `mod.rs` because that's what I'm used to from the time when there was no other option but if there already exists non-directory `foo.rs` and I need to add  files I skip renaming it.

If we want to have a policy about this (I'm lightly against) then `foo.rs` is better than `foo/mod.rs` because it's easier to add things.